File: gradle-module-plugin/src/functionalTest/resources/designer/DesignerHook.java
package le.examp;

import com.inductiveautomation.ignition.common.licensing.LicenseState;
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook;
import com.inductiveautomation.ignition.designer.model.DesignerContext;


/**
 * This is the Designer-scope module hook used for tests.
 */
public class DesignerHook extends AbstractDesignerModuleHook {

    public static final String MODULE_ID = "component-example";

    @Override
    public void startup(DesignerContext context, LicenseState activationState) throws Exception {
        // noop
    }
}

File: module-generator/core/src/main/resources/templates/hook/DesignerHook.kt
package <PACKAGE_ROOT>.designer

import com.inductiveautomation.ignition.common.licensing.LicenseState
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook
import com.inductiveautomation.ignition.designer.model.DesignerContext


/**
 * This is the Designer-scope module hook.  The minimal implementation contains a startup method.
 */
class <MODULE_CLASSNAME>DesignerHook: AbstractDesignerModuleHook() {

    // override additonal methods as requried

    @Throws(Exception)
    override fun startup(context: DesignerContext, activationState: LicenseState) {
        // implelement functionality as required
    }
}
